test/Parse/collection_conditionals_error.swift

```diff
@@ -0,0 +1,53 @@
+// RUN: %target-typecheck-verify-swift
+
+[ // expected-warning {{expression of type '[String]' is unused}}
+  #if FLAG
+  "foo",
+  #else
+  "foo": 12, // expected-error {{expected ',' separator}}
+  // expected-error@-1{{expected expression in container literal}}
+  #endif
+]
+
+[ // expected-warning {{expression of type '[String : Int]' is unused}}
+#if FLAG
+  // FIXME: commenting out until blah blah...
```

Comment:
Please remove this comment. It's confusing for the test case.

Reply:
Done.